### DISCUSSION TITLE
Sync variables between two players

### DISCUSSION THREAD

**Discussion by EwertonLug** [inkle/ink#900] [General]
I know it wasn't designed to work in multiplayer games. But I'm going to work on a multiplayer game using it and I would like to know what would be the best way to synchronize global variables between two players.

**For example.**
If player 1 opens a door, player 2 will have the door open when they reach that part.

**Comment by Selsynn** [comment]
So first, you can't do it from ink side, it needs to be handled with the rest of your game client, outside of ink, so in the prog language you are using outside. 
For the example, I'll suppose you are using C#. Here are the steps I'll do:
- identify and make the variable sync easy to differentiate from the variable not synced. Created at the same place in the ink code for example, in an include about player sync. I would also namespace the variable for when you use it, you know that there is the sync to think of. like "sync_door_state" instead of having a "open_door" boolean. Everything that begins with "sync" could potentially change state at any point, without any specific action from the ink side. Not every variable has to by synced, for example if you are tracking the character relationship with some npc. The position of the npc may be shared, but how much the npc likes or dislike this player may not be shared.
- For each of the synced variable, implement getter/setter in the code. (see https://github.com/inkle/ink/blob/master/Documentation/RunningYourInk.md#settinggetting-ink-variables)
- I assume you already have a system running that will run some code when the other is updating a synced variable, in that hook, you want to update the variable in the inkstate, and you want to hot reload too, if the scene changed. (to update the variable will be with the previous link. I'm not exactly sure how to hot-reload. A painful way would be to store where you are in the ink story and then jump to that particular scene (with https://github.com/inkle/ink/blob/master/Documentation/RunningYourInk.md#jumping-to-a-particular-scene) but I'm not sure if there is an easier way

Good luck, and let me know if you run into any problems

**Reply by EwertonLug** [reply]
Thank you for the explanation.
